### PR TITLE
fix podman logs --tail when log is bigger than pagesize

### LINKED ISSUE #7230: podman logs often cannot use --tail due to some parsing issue
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

When I try to look at the logs of a running container using --tail, I often get back some error about having trouble parsing the log.

**Steps to reproduce the issue:**

1.  I have a running mariadb container

2.  If I issue the following (at the moment):
```
podman logs --tail 10 mariadb
```
I get:

```
2020-08-05  1:34:35 0 [Note] InnoDB: File './ibtmp1' size is now 12 MB.
2020-08-05  1:34:35 0 [Note] InnoDB: 10.5.4 started; log sequence number 891100233; transaction id 336651
2020-08-05  1:34:35 0 [Note] InnoDB: Loading buffer pool(s) from /var/lib/mysql/ib_buffer_pool
2020-08-05  1:34:35 0 [Note] Plugin 'FEEDBACK' is disabled.
2020-08-05  1:34:35 0 [Note] Server socket created on IP: '::'.
2020-08-05  1:34:35 0 [Note] InnoDB: Buffer pool(s) load completed at 200805  1:34:35
2020-08-05  1:34:35 0 [Note] Reading of all Master_info entries succeeded
2020-08-05  1:34:35 0 [Note] Added new Master_info '' to hash table
2020-08-05  1:34:35 0 [Note] mysqld: ready for connections.
Version: '10.5.4-MariaDB-1:10.5.4+maria~focal-log'  socket: '/run/mysqld/mysqld.sock'  port: 3306  mariadb.org binary distribution
```

3. However if I issue the following (at the moment as the log changes):
```
podman logs --tail 11 mariadb
```
I get:
```
Error: unable to read log file 08b28134eab00acd1ed427500cf4eb00c2fad67c6af6ffca106c616117f94b96 for /var/home/core/.local/share/containers/storage/overlay-containers/08b28134eab00acd1ed427500cf4eb00c2fad67c6af6ffca106c616117f94b96/userdata/ctr.log : unable to convert time ile from container log: parsing time "ile" as "2006-01-02T15:04:05.000000000Z07:00": cannot parse "ile" as "2006"
```
The 11th line back if I just do ```podman logs mariadb``` is:
```
2020-08-05  1:34:35 0 [Note] InnoDB: Setting file './ibtmp1' size to 12 MB. Physically writing the file full; Please wait ...
```

**Describe the results you received:**

```
Error: unable to read log file 08b28134eab00acd1ed427500cf4eb00c2fad67c6af6ffca106c616117f94b96 for /var/home/core/.local/share/containers/storage/overlay-containers/08b28134eab00acd1ed427500cf4eb00c2fad67c6af6ffca106c616117f94b96/userdata/ctr.log : unable to convert time ile from container log: parsing time "ile" as "2006-01-02T15:04:05.000000000Z07:00": cannot parse "ile" as "2006"
```

**Describe the results you expected:**

No errors and just the logs to be output showing the number of lines specified.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `podman version`:**

```
Version:            1.9.3
RemoteAPI Version:  1
Go Version:         go1.14.2
OS/Arch:            linux/amd64
```

**Output of `podman info --debug`:**

```
debug:
  compiler: gc
  gitCommit: ""
  goVersion: go1.14.2
  podmanVersion: 1.9.3
host:
  arch: amd64
  buildahVersion: 1.14.9
  cgroupVersion: v1
  conmon:
    package: conmon-2.0.18-1.fc32.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.0.18, commit: 6e8799f576f11f902cd8a8d8b45b2b2caf636a85'
  cpus: 12
  distribution:
    distribution: fedora
    version: "32"
  eventLogger: file
  hostname: coreos.eacompany.net
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
  kernel: 5.7.8-200.fc32.x86_64
  memFree: 3522252800
  memTotal: 67493203968
  ociRuntime:
    name: runc
    package: runc-1.0.0-144.dev.gite6555cc.fc32.x86_64
    path: /usr/bin/runc
    version: |-
      runc version 1.0.0-rc10+dev
      commit: fbdbaf85ecbc0e077f336c03062710435607dbf1
      spec: 1.0.1-dev
  os: linux
  rootless: true
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.1.1-1.fc32.x86_64
    version: |-
      slirp4netns version 1.1.1
      commit: bbf27c5acd4356edb97fa639b4e15e0cd56a39d5
      libslirp: 4.3.1
      SLIRP_CONFIG_VERSION_MAX: 2
  swapFree: 0
  swapTotal: 0
  uptime: 104h 22m 4.61s (Approximately 4.33 days)
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - registry.centos.org
  - docker.io
store:
  configFile: /var/home/core/.config/containers/storage.conf
  containerStore:
    number: 19
    paused: 0
    running: 19
    stopped: 0
  graphDriverName: overlay
  graphOptions:
    overlay.mount_program:
      Executable: /usr/bin/fuse-overlayfs
      Package: fuse-overlayfs-1.1.2-1.fc32.x86_64
      Version: |-
        fusermount3 version: 3.9.1
        fuse-overlayfs: version 1.1.0
        FUSE library version 3.9.1
        using FUSE kernel interface version 7.31
  graphRoot: /var/home/core/.local/share/containers/storage
  graphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "false"
  imageStore:
    number: 75
  runRoot: /run/user/1000/containers
  volumePath: /var/home/core/.local/share/containers/storage/volumes
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
podman-1.9.3-1.fc32.x86_64
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
Fedora CoreOS

### PR DESCRIPTION
**Pull request by Luap99**
Fixes #7230

### CODE REVIEW

File: libpod/logs/reversereader/reversereader.go [hunk 1]
```diff
@@ -61,6 +61,6 @@ func (r *ReverseReader) Read() (string, error) {
 		b = b[0:n]
 	}
 	// Set to the next page boundary
-	r.offset = -r.readSize
+	r.offset -= r.readSize
```
Comment (giuseppe) [inline libpod/logs/reversereader/reversereader.go:64]: let's adjust the comment to reflect what is done now

**Comment by rhatdan** [comment]
LGTM with @giuseppe comment.

**Comment by mheon** [comment]
New test also seems to be failing

**Comment by TomSweeneyRedHat** [comment]
LGTM
assuming happy tests and the code comment is tweaked.

**Comment by Luap99** [comment]
OK, I updated the code. Tests were failing due an race condition were the file was closed to early. Should be fixed now.

File: libpod/logs/log.go [hunk 2]
```diff
@@ -101,11 +101,14 @@ func getTailLog(path string, tail int) ([]*LogLine, error) {
 			if err != nil {
 				if errors.Cause(err) == io.EOF {
 					inputs <- []string{leftover}
-					close(inputs)
```
Comment (rhatdan) [inline libpod/logs/log.go:104]: Is this close done somewhere else?  Is it not needed?

Comment (Luap99) [reply libpod/logs/log.go:104]: It is closed below

**Comment by rhatdan** [comment]
/approve
Thanks @Luap99

**Pull request merged**